nerfstudio/models/nerfacto.py

```diff
@@ -345,14 +333,13 @@ def get_metrics_dict(self, outputs, batch):
     def get_loss_dict(self, outputs, batch, metrics_dict=None):
         loss_dict = {}
         image = batch["image"].to(self.device)
-        blended_rgb = (
-            self.renderer_rgb.blend_background(
-                rgb=image[..., :3], opacity=image[..., 3], background_color=outputs["background_color"]
-            )
-            if image.shape[-1] == 4 and "background_color" in outputs
-            else image[..., :3]
+        pred_rgb, gt_rgb = self.renderer_rgb.blend_background_for_loss_computation(
```

Comment:
Wouldn't this add the background color twice if the background color isn't random? First time in `combine_rgb`.

Reply:
What about removing it from `combine_rgb` and making sure that we do this through one of the other methods if required? It is more of a trick rather than being part of volume rendering anyways.

Reply:
Yes, you are right! It was adding the background color twice.

Reply:
Do you mean moving it to `forward`? We would need to introduce the "add_background" method (as blending cannot be used for the case).

Reply:
I was thinking in terms of having one function, `blend_background_for_metric_computation` and one `blend_background_for_loss_computation`. One of them would always be called and would do the background color adding. Having it in `forward` wouldn't prevent the adding twice problem.

If we don't want a `blend_background_for_metric_computation` method, we could add a flag to `forward` to determine whether to add or not.

Reply:
Is this still an issue or was it resolved?

Reply:
Resolved with https://github.com/nepfaff/nerfstudio/pull/1#discussion_r1265211859